code/model/modelreplace.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "globalincs/vmallocator.h"
+#include "model/model.h"
+
+#include <memory>
+
+bool model_exists(const SCP_string& filename);
+
+bool read_virtual_model_file(polymodel* pm, const SCP_string& filename, model_parse_depth depth, int ferror, model_read_deferred_tasks& deferredTasks);
```

Comment:
Since virtual POF parsing is basically depth-first searching through the dependency tree, every layer needs to keep a copy of the depth for the next branch of the tree. While the depth needs to propagate downwards the dependency tree, it must not propagate to sibling branches or parents.
Consider the following example:

Virtual pof A replaces model A
Virtual pof B uses model A
Virtual pof C uses model B and model A

If you now try to load model C, the following resolve will happen:
Load virtual model C.
Virtual C loads virtual model B.
Virtual B loads virtual model A (depth = 0).
Virtual A loads real model A (depth = 1).
Virtual C loads virtual model A (depth = 0).
Virtual A loads real model A (depth = 1).

These last two steps are the issue. If depth is passed by reference, then the depth for model A during the "C loads virtual A" step will be 2 (since it would have been incremented the first time we loaded the two A models) instead of the correct 0.